include/deal.II/lac/generic_linear_algebra.h

```diff
@@ -154,10 +229,68 @@ namespace LinearAlgebraTrilinos
 {
   using namespace dealii;
 
+  /**
+   * Typedef for the base class for preconditioners.
+   *
+   * This typedef is useful when wanting to create a generic solver interface
+   * wherein one can switch between preconditioners. For example:
+   * @code
+   * std_cxx11::unique_ptr<LA::PreconditionerBase> preconditioner;
+   * if (parameters.preconditioner_type == "Jacobi") {
+   * LA::MPI::PreconditionJacobi* ptr_prec
+   *   = new LA::MPI::PreconditionJacobi ();
+   * ...
+   * preconditioner.reset(ptr_prec);
```

Comment:
Why split it into a pointer assignment followed by a unique-ptr assignment, btw?